### PR TITLE
[Bug fix] adjust auth error message for better understanding

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java [hunk 1]
```diff
@@ -387,7 +384,7 @@ private static void promptForOAuthOrDeviceCodeLogin(AuthType authType) {
     private static Mono<Account> findFirstAvailableAccount() {
         final List<Account> accounts = Azure.az(AzureAccount.class).accounts();
         if (accounts.isEmpty()) {
-            return Mono.error(new AzureToolkitAuthenticationException("There are no accounts available."));
+            return Mono.error(new AzureToolkitAuthenticationException("there are no subscriptions available."));
```
Comment (shenqianjin) [inline azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java:387]: there is no subscription ?

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/util/AzureCliUtils.java [hunk 2]
```diff
@@ -32,12 +32,12 @@ public static void ensureMinimumCliVersion() {
...
             // we require at least azure cli version 2.11.0
             if (compareWithMinimVersion(cliVersion) < 0) {
-                throw new AzureToolkitAuthenticationException(String.format("Your azure cli version '%s' is too old, " +
+                throw new AzureToolkitAuthenticationException(String.format("your azure cli version '%s' is too old, " +
                         "you need to upgrade your CLI with 'az upgrade'.", cliVersion));
             }
         } catch (NullPointerException | NumberFormatException ex) {
             throw new AzureToolkitAuthenticationException(
-                    String.format("Cannot authenticate through azure cli, " +
+                    String.format("azure cli is not ready, " +
```
Comment (shenqianjin) [inline azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/util/AzureCliUtils.java:40]: "Azure CLI"

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java (hunk 1 above)
Comment (andxu) [reply azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java:387]: both are correct

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/AzureAccount.java [hunk 3]
```diff
@@ -252,15 +251,15 @@ protected String getClientId() {
     }
 
     private static Region toRegion(com.azure.core.management.Region region) {
-        return Optional.ofNullable(Region.fromName(region.name())).orElseGet(() -> new Region(region.name(), region.label() + "*"));
+        return Optional.of(Region.fromName(region.name())).orElseGet(() -> new Region(region.name(), region.label() + "*"));
```
Comment (wangmingliang-ms) [inline azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/AzureAccount.java:254]: seems can `return Region.fromName(region.name())` directly

**Review by wangmingliang-ms — APPROVED** [review]

**Pull request merged**